src/GslibIO.jl

```diff
@@ -52,28 +79,42 @@ Optionally set the value used for missings `na`.
 """
 function load_legacy(filename::AbstractString, dims::NTuple{3,Int};
                      origin=(0.,0.,0.), spacing=(1.,1.,1.), na=-999)
-  open(filename) do fs
-    # skip header
-    readline(fs)
+  metadata, data = parse_legacy(filename)                     
 
-    # number of properties
-    nvars = parse.(Int, readline(fs))
+  # handle missing values
+  replace!(data, na=>NaN)
 
-    # properties names
-    vars = [Symbol(readline(fs)) for i in 1:nvars]
+  # create data dictionary
+  table = (; zip(metadata.varnames, eachcol(data))...)
+  domain = RegularGrid(dims, origin, spacing)
 
-    # read property values
-    X = readdlm(fs)
+  georef(table, domain)
 
-    # handle missing values
-    replace!(X, na=>NaN)
+end
 
-    # create data dictionary
-    data = (; zip(vars, eachcol(X))...)
-    domain = RegularGrid(dims, origin, spacing)
+"""
+    load_legacy(filename, coordnames, na=-999)
 
-    georef(data, domain)
-  end
+Load legacy GSLIB `filename` into a PointSet using the properties in `coordnames` as coordinates.
+Optionally set the value used for missings `na`.
+"""
+function load_legacy(filename::AbstractString, coordnames=(:x, :y, :z); na=-999)
+  metadata, data = parse_legacy(filename)
+
+  # handle missing values
+  replace!(data, na=>NaN)
+
+  # create temporary data dictionary to split later
+  # coordinates and attributes
+  tableall = Dict(zip(metadata.varnames, eachcol(data)))
+
+  # create data for coordinates
+  coords = transpose(reduce(hcat, [tableall[c] for c in coordnames]))
+  # create table with varnames not in coordnames
+  attrnames = [c for c in metadata.varnames if c ∉  coordnames]
+  table = (; zip(attrnames, [tableall[c] for c in attrnames])...)
```

Comment:
I think we can simplify this code further. You have the `data` table and the `metadata.varnames` in order. So I would just find the indices of the `coordnames` in `metadata.varnames` and then do `coords = data[:,inds]'` to collect the coordinates. The remaining indices are the variables, which you can convert to a table using `(; zip(metadata.varnames[other], eachcol(data[:,other]))` or something.

I feel that we need to review the LegacySpec struct to reflect the fact that some of the `varnames` can actually be coordnames. This is a bit confusing still for someone reading the code for the first time. Do you have suggestions on how to improve this?

Reply:
> I think we can simplify this code further. You have the `data` table and the `metadata.varnames` in order. So I would just find the indices of the `coordnames` in `metadata.varnames` and then do `coords = data[:,inds]'` to collect the coordinates. The remaining indices are the variables, which you can convert to a table using `(; zip(metadata.varnames[other], eachcol(data[:,other]))` or something.

I agree. This proposed implementation is using the same coding principles that in the existing load_legacy, but surely I can write it as proposed.

Reply:
To find the index in the vector, you can use the helpful `indexin` function

Reply:
> I feel that we need to review the LegacySpec struct to reflect the fact that some of the `varnames` can actually be coordnames. This is a bit confusing still for someone reading the code for the first time. Do you have suggestions on how to improve this?

If you check the GSLIB specification you will realise that there is not way to establish which variables are coordinates. It is responsibility of the user to indicate them.

Reply:
Yes, I know. I just feel that we need to improve upon this strange and incomplete specification. We need to show readers of the code that this is a possibility, and that the code will handle variables that are actually coordinates later on.

Reply:
Agree, but doing so it means changing the GSLIB format which is not the goal of this PR.

Reply:
How are we changing the GSLIB format? We are just documenting what these variables are. I am not proposing to change anything in the struct other than the comments attached to the `varnames` field.

Reply:
If you check the entire GeoStats.jl stack, the identifer `varnames` is never used for `coordnames` so if someone comes here and sees `varnames` they will likely assume the incorrect definition. That is why we need to clarify this with comments well. Makes sense?

Reply:
> If you check the entire GeoStats.jl stack, the identifer `varnames` is never used for `coordnames` so if someone comes here and sees `varnames` they will likely assume the incorrect definition. That is why we need to clarify this with comments well. Makes sense?

I don't understand what you are asking for. Apparently you ask for better documentation now? At first, I thought your were asking to add extra something in the structure. Could you clarify this?

Reply:
I meant clarification in the form of documentation. No need to change the struct other than the modification about the `data` field.